plugins/woo-ai/src/utils/productData.ts

```diff
@@ -1,32 +1,25 @@
 /**
  * Internal dependencies
  */
-import { Attribute, ProductData } from './types';
-import { getTinyContent, getPostId } from '.';
+import { Attribute } from './types';
+import { getTinyContent } from '.';
 
-const isElementVisible = ( element: HTMLElement ) =>
-	! ( window.getComputedStyle( element ).display === 'none' );
-
-const getCategories = (): string[] => {
-	const categoryCheckboxEls: NodeListOf< HTMLInputElement > =
+export const getCategories = () =>
```

Comment:
If we can change this always to return an array of strings (and not undefined), it would be more predictable. It can return an empty array if nothing is found. Something like this:

```typescript
export const getCategories = (): string[] => {
	const categoryInputs: NodeListOf< HTMLInputElement > =
		document.querySelectorAll< HTMLInputElement >(
			'#taxonomy-product_cat input[name="tax_input[product_cat][]"]'
		);

	if ( ! categoryInputs.length ) {
		// if no results, return an empty array
		return [];
	}

	return Array.from( categoryInputs )
		.filter(
			( inputElement: HTMLInputElement ) =>
				window.getComputedStyle( inputElement, ':before' ).content !==
				'none'
		)
		.map(
			( inputElement: HTMLInputElement ) =>
				inputElement.nextSibling?.nodeValue?.trim() ?? ''
		);
};
```

Reply:
Good call! Done in 421f3534539d5bc3f2bde5aaabc9eb51e4c91ad8